File: melosys-eessi-app/src/test/java/no/nav/melosys/eessi/service/journalfoering/JournalpostMetadataServiceTest.java
package no.nav.melosys.eessi.service.journalfoering;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.assertj.core.api.Assertions.assertThatExceptionOfType;

@ExtendWith(MockitoExtension.class)
class JournalpostMetadataServiceTest {
    private final JournalpostMetadataService journalpostMetadataService = new JournalpostMetadataService();

    @Test
    void hentJournalpostMetadata_ugyldigSedType_kasterFeil() {
        assertThatExceptionOfType(IllegalArgumentException.class)
            .isThrownBy(() -> journalpostMetadataService.hentJournalpostMetadata("ugyldig SedType"))
            .withMessageContaining("No enum constant");
    }
}

File: melosys-eessi-app/src/test/java/no/nav/melosys/eessi/service/journalfoering/JournalpostMetadataServiceTest.kt
package no.nav.melosys.eessi.service.journalfoering

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith

import io.kotest.assertions.throwables.shouldThrow
import io.kotest.matchers.string.shouldContain
import io.mockk.junit5.MockKExtension
import java.lang.IllegalArgumentException

@ExtendWith(MockKExtension::class)
internal class JournalpostMetadataServiceTest {
    private val journalpostMetadataService = JournalpostMetadataService()

    @Test
    fun hentJournalpostMetadata_ugyldigSedType_kasterFeil() {
        shouldThrow<IllegalArgumentException> {
            journalpostMetadataService.hentJournalpostMetadata("ugyldig SedType")
        }.message shouldContain "No enum constant"
    }
}
